spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -736,6 +736,10 @@ def log_lsp_handle_errors(self, message):
             # there. So we intentionally leave an error in this call
             # to get the entire stack info generated by it, which
             # gives the info we need from users.
+            if PY2 and error:
+                # For Python 2 we need the actual error so we can raise it
+                raise error

```

Comment:
The whole point of logging LSP handling errors, instead of raising them, is that after they are raised, then code completion, linting and all other services provided by LSP servers can't be handled by Spyder anymore.

I just tested this with your solution and it happens for Python 2 as it was happening for both Python 2 and 3 before PR #9138. So my proposed solution is this:

```python
if PY2:
    logger.error(message, exc_info=True)
    print(message, file=sys.stderr)
else:
    logger.error('%', 1, stack_info=True)
```

It doesn't show as much info as `stack_info`, but it's informative enough. Besides, given that Python 2 downloads have decreased a lot in the last year (just 18%, according to Anaconda statistics), I think we should not waste too much time on this.